### PR TITLE
[DOCS-2840] Update Go and Java log collection docs

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do?

For the Go log collection doc:
- Add more instructions for Configure the Datadog Agent section.
- Move the Connect Traces and Logs section to after Configure the Datadog Agent
- Copy edits

Java log collection doc:
- Add missing paragraph.

### Motivation

DOCS-2840

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/logs/log_collection/go.md [hunk 1]
```diff
@@ -78,38 +76,53 @@ func main() {
 }
 ```
 
-**Connect Logs and Traces**
+## Configure your Datadog Agent
 
-If APM is enabled for this application, the correlation between application logs and traces can be improved by [following APM Go logging instructions][3] to automatically add trace and span IDs in your logs.
+Once [log collection is enabled][3], set up [custom log collection][4] to tail your log files and send them to Datadog.
```
[the remark this answers is not in the document]
Comment (buraizu) [reply content/en/logs/log_collection/go.md:81]: Optional suggestion to clarify which logs will be sent (based on questions I've had from customers in the past).

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks, just a few minor points, and also asked a question

File: content/en/logs/log_collection/go.md [hunk 2]
```diff
@@ -78,38 +76,53 @@ func main() {
...
+    ```
 
-  - type: file
-    path: "/path/to/your/go/log.log"
-    service: go
-    source: go
-    sourcecategory: sourcecode
-```
+3. [Restart the Agent][6].
+4. Run the [Agent’s status subcommand][7] and look for `csharp` under the `Checks` section to confirm logs are successfully submitted to Datadog.
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/logs/log_collection/go.md:98]: Lol ooops...caught copying and pasting. Thanks for catching this!

Comment (maycmlee) [inline content/en/logs/log_collection/go.md:98]:
```suggestion
4. Run the [Agent’s status subcommand][7] and look for `go` under the `Checks` section to confirm logs are successfully submitted to Datadog.
```

**Review by buraizu — APPROVED** [review]

**Pull request merged**